model-engine/tests/unit/infra/gateways/test_s3_llm_artifact_gateway.py

```diff
@@ -0,0 +1,81 @@
+from typing import List
+from unittest import mock
+
+import pytest
+from model_engine_server.common.config import hmi_config
+from model_engine_server.infra.gateways.s3_llm_artifact_gateway import S3LLMArtifactGateway
+
+
+@pytest.fixture
+def llm_artifact_gateway():
+    gateway = S3LLMArtifactGateway()
+    return gateway
+
+
+@pytest.fixture
+def fake_files():
+    return ["fake-prefix/fake1", "fake-prefix/fake2", "fake-prefix/fake3"]
+
+
+def mock_boto3_session(fake_files: List[str]):
+    mock_session = mock.Mock()
```

Comment:
`moto` could be an option here, though this mocking can be fine too since you're already at the last layer, where you're in the S3 artifact gateway (want to avoid mocking across multiple layers).

Reply:
ah moto seems cool, useful to know for the future. wanted to create a custom side effect so this leans itself more easily to that